Current Version Display for Questions and Response Sets

kierk opened the pull request:
This PR adds a display of the current version and version currently selected on the show page for both response sets and questions. Please review how it looks and let me know any stylistic changes that should be made.

Make sure you have checked off the following before you issue your Pull Request:

- [N/A] Added unit tests for new functionality
- [x] Passed all unit tests using `rails test` with 90%+ coverage
- [N/A] Added cucumber tests for any new functionality
- [x] Passed all cucumber tests using `bundle exec cucumber`
- [x] Passed overcommit hooks, including running all code through Rubocop
- [N/A] If any database changes were made, run `rake generate_erd` to update the README.md

eedrummer commented:
I think this will work, but in my opinion, there is too much logic going on in the views. A lot of it seems to be there to accommodate how the other_versions method on ResponseSet/Question works. I wrote that method to support the UI as is, but as our last sprint review showed, that UI doesn't show all of the information necessary to support the user. My recommendation would be to refactor that method into something like "all_versions". It should be easy (remove the where.not clause). The views can then have a simple loop that just compares ov.version to @response_set.version to write out current version into the list.

I'd also recommend adding methods to the ResponseSet/Question instances to see if they are the most recent version and to return the most recent version number. We want the views to ask the models for the information that they need. We don't want the views to have to figure that sort of thing out.

kierk commented:
I moved the logic to the models - thanks for the suggestion @eedrummer, it was much easier to do it the way you were suggesting / with the suggested methods. I rebased and retested everything. Let me know if you see any other issues.

eedrummer commented:
This looks much more clean. The only issue that I see is somehow the authentications table made it into schema changes. I don't think this PR requires any changes to the DB. Could you just drop the changes to schema.rb from the PR?

kierk commented:
@eedrummer schema.rb changes are no longer part of the PR
